### PR TITLE
Implement in-place editing for some of the project's fields

### PR DESCRIPTION
**Pull request by danidoni**
Use in-place editing for the project's
- Title
- Url
- Description

We've been using modals to edit the project's title, url and description. We
think we have better tools for that. In-place editing is one of the ways to do
it.

With this PR we introduce a new way of updating some of the project's
information from the projects overview page.

Co-authored-by: Saray Cabrera Padrón <scabrerapadron@suse.de>

Here is a screenshot of how it looks:

** Before **
Screenshot of the project monitor

** After **
Screenshot of the project monitor

### CODE REVIEW

File: src/api/app/views/webui/package/show.html.haml [hunk 1]
```diff
@@ -93,3 +93,10 @@
     = render partial: 'webui/request/delete_request_dialog', locals: { project: @project, package: @package }
   - if @package.develpackage
     = render partial: 'webui/request/change_devel_request_dialog', locals: { project: @project, package: @package }
+
+- content_for :ready_function do
+  :plain
+    $(document).ready(function() {
```
Comment (coolo) [inline src/api/app/views/webui/package/show.html.haml]: :ready_function is already in the ready function

**Comment by danidoni** [comment]
[Modals](https://trello.com/c/cq2te0jh/1332-modals)

File: src/api/app/views/webui/package/show.html.haml (hunk 1 above)
Comment (danidoni) [reply src/api/app/views/webui/package/show.html.haml]: I'm gonna remove this, but thanks

Comment (danidoni) [reply src/api/app/views/webui/package/show.html.haml]: I mean, this is left over stuff from a previous experiment

File: src/api/app/controllers/webui/project_controller.rb [hunk 2]
```diff
@@ -223,13 +223,22 @@ def restore
...
-        flash[:error] = 'Failed to update project'
+      format.html do
+        if request.xhr?
+          render partial: "edit_project", locals: { project: @project }, layout: false, status: :ok
+        else
+          redirect_to project_show_path(@project)
+        end
       end
-      format.html { redirect_to project_show_path(@project) }
+      format.js { render json: @project, status: :ok }
```
Comment (vpereira) [inline src/api/app/controllers/webui/project_controller.rb]: we don't render json unless we use it to Datatables

Comment (danidoni) [reply src/api/app/controllers/webui/project_controller.rb]: True, removing

**Comment by danidoni** [comment]
I'll close this PR in favor of https://github.com/openSUSE/open-build-service/pull/9203 as we're shifting from editing fields one by one to editing them at once in an in-place form

**Comment by danidoni** [comment]
[Modals](https://trello.com/c/cq2te0jh/1332-modals)

**Pull request closed without merging**